### PR TITLE
Update POTCAR summary stats to include 6.4 POTCARs and add `dev_script` utils for future updates

### PR DESCRIPTION
**Pull request by esoteric-ephemera**
### Major changes:

- Follow-up to issue [#3342](https://github.com/materialsproject/pymatgen/issues/3342) and closed PR [#3351](https://github.com/materialsproject/pymatgen/pull/3351)
- Adds support for version 64 POTCARs
- Adds hidden function for PMG devs to continuously update potcar_summary_stats.json.gz used to validate potcars, `pymatgen.io.vasp.inputs._gen_potcar_summary_stats`
- Added functionality to generate fake POTCARs from real POTCARs (by randomizing numeric and boolean values in them) for tests in `dev_scripts/potcar_scrambler.py`

### CODE REVIEW

File: pymatgen/io/vasp/inputs.py [hunk 1]
```diff
@@ -2280,6 +2283,43 @@ def __repr__(self) -> str:
         return f"{cls_name}({symbol=}, {functional=}, {TITEL=}, {VRHFIN=}, {n_valence_elec=:.0f})"
 
 
+def _gen_potcar_summary_stats():
+    """
+    This function solely intended to be used for PMG development to regenerate the
+    potcar_summary_stats.json.gz file used to validate POTCARs
+    """
```
Comment (janosh) [inline pymatgen/io/vasp/inputs.py:2290]: Perfect! This is great to have. Even though it's marked as private, we should add a test for this.

**Review by janosh — CHANGES_REQUESTED** [review]

File: pymatgen/io/vasp/inputs.py [hunk 2]
```diff
@@ -2280,6 +2283,43 @@ def __repr__(self) -> str:
         return f"{cls_name}({symbol=}, {functional=}, {TITEL=}, {VRHFIN=}, {n_valence_elec=:.0f})"
 
 
+def _gen_potcar_summary_stats():
```
Comment (mkhorton) [inline pymatgen/io/vasp/inputs.py:2286]: This is fine here, but noting there is also a top-level `dev_scripts` folder (largely undocumented). On balance it's probably better here for discoverability for anyone curious how this file was generated.

Minor comments: the bare `return` is unnecessary, and `monty.serialization.dumpfn` can be used directly with `dumpfn(new_summary_stats, "filename.json.gz")` which is a convenience function we use a lot.

Comment (esoteric-ephemera) [reply pymatgen/io/vasp/inputs.py:2286]: Thanks! Much better way of doing this. For placement, you and @janosh probably have a better sense of where this belongs. `_gen_potcar_summary_stats` is a "destructive" function, in that it overwrites the file that's distributed with PMG. `dev_scripts` might then be a better place for it

File: pymatgen/io/vasp/inputs.py (hunk 1 above)
Comment (esoteric-ephemera) [reply pymatgen/io/vasp/inputs.py]: Added a `Test_gen_potcar_summary_stats` to `tests/io/vasp/test_inputs.py`. Requires generating a library of fake POTCARs, which are in `tests/fake_POTCAR_library`. Code to generate this library from a real library of POTCARs is in `dev_scripts/potcar_scrambler.py`

**Review by janosh — APPROVED** [review]
Really nice work, thanks @esoteric-ephemera! 👍

**Comment by lan496** [comment]
Hi. I think `pymatgen/io/vasp/potcar_summary_stats.json.gz` should be added to `setup.package_data` to use it via pip-installed version.

https://github.com/materialsproject/pymatgen/blob/44bf38318f1ff4265385eb6587bb713780732ff7/setup.py#L95-L119

**Comment by janosh** [comment]
Ah crap!

**Comment by mkhorton** [comment]
Not the first time don’t worry! I once added a catch-all to package all .json, .yaml files by default but this was vetoed for reasons I forget, but perhaps worth reconsidering

**Comment by janosh** [comment]
> I once added a catch-all to package all .json, .yaml files by default but this was vetoed for reasons I forget, but perhaps worth reconsidering

I think it's worth a try. @shyuep Any concerns?

If we get false positives, we can always revert the change.

**Comment by shyuep** [comment]
My concern is that data files should be few and far between and EXPLICIT.

**Comment by janosh** [comment]
> My concern is that data files should be few and far between and EXPLICIT.

Exactly, which is why the ones we deliberately place in the pymatgen pkg dir should be there for a good reason and hence also be included in the PyPI release. At least that's the case right now as you can see from [d8151e5](https://github.com/materialsproject/pymatgen/pull/3374/commits/d8151e5de2c0c90ed5b28375d21bc573dd778dfd) passing.

**Comment by shyuep** [comment]
Yes, that is why it should be explicitly specified. For the same reason, you do not usually do `from package import *`.

**Comment by janosh** [comment]
The biggest reason not to do `from package import *` imo is that it breaks tooling. IDEs can trace where things come from so no doc string glancing, no type hints and worse code completions. But none of those apply here, there's no tooling for pkg data anyway. So far, there's been a 1-to-1 map from data files in `pymatgen` to things included in `SOURCES.txt` so we might as well make the map automatic imo.

**Comment by shyuep** [comment]
There are reasons. Say someone decides to add a `very_big_file.yaml` for some internal testing purposes. That file will automatically be added to package data, even if it has no real purpose in actual function of the main code. Like I said, it is better to be explicit what files are needed and what are not.

**Comment by janosh** [comment]
I don't merge very big files into `master`. Not even in test files, let alone in pkg dir. Test files should never be placed in pkg dir.

**Pull request merged**